app/scripts/controllers/createClientBuild.js

```diff
@@ -131,6 +131,14 @@ angular.module('openshiftConsole')
         };
       }
 
+      if (clientConfig.envVars) {
+        _.forEach(clientConfig.envVars, function(envVar) {
+          if (envVar.name) {
+            buildConfig.spec.strategy.jenkinsPipelineStrategy.env.push(JSON.parse(angular.toJson(envVar)));
```

Comment:
@sedroche Needed to parse after doing `angular.toJson` as this returns the object as a string.